File: front/src/services/UserService.js
import http from "../services/HttpService";

class UserService{
    async exists(username, password){
        return await http.post("/user/exists", {
            username,
            password
        });
    }

    async save(user){
        return await http.post("/user", user);
    }

    async update(){
        return await http.put("/user", {});
    }

    async findAll(){
        return await http.get("/user");
    }

    async find(username){
        return await http.get(`/user/${username}`);
    }
}

export const service = new UserService();
export default service;

File: back/services/UserService.py
from __main__ import app
from flask import request
from repositories.UserRepository import UserRepository
from utils import wrapper
from utils import md5

repository = UserRepository()

@app.get("/user")
def findAll():
    users = repository.findAll()
    return wrapper(users)

@app.get("/user/<id>")
def findById(id):
    user = repository.find(id)
    return wrapper(user)

@app.post("/user/checkUser")
def checkUser():
    user = request.get_json(silent=True)
    username = user.get("username")
    password = md5(user.get("password"))
    user = repository.checkUser(username, password)
    return wrapper(user)

@app.post("/user")
def saveUser():
    user = request.get_json(silent=True)
    response = repository.save(user)
    return wrapper(response)

@app.put("/user")
def updatePassword():
    user = request.get_json(silent=True)
    response = repository.updatePassword(user)
    return wrapper(response)

